File: guess-game-server/src/main/java/guess/service/AnswerService.java
package guess.service;

import guess.domain.answer.ErrorDetails;
import guess.domain.answer.Result;

import javax.servlet.http.HttpSession;
import java.util.List;

/**
 * Answer service.
 */
public interface AnswerService {
    void setAnswer(int questionIndex, long answerId, HttpSession httpSession);

    int getCurrentQuestionIndex(HttpSession httpSession);

    List<Long> getWrongAnswerIds(int questionIndex, HttpSession httpSession);

    Result getResult(HttpSession httpSession);

    List<ErrorDetails> getErrorDetailsList(HttpSession httpSession);
}

File: guess-game-web/src/app/shared/services/answer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { MessageService } from '../../modules/message/message.service';
import { Observable } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { Result } from '../models/result/result.model';

@Injectable({
  providedIn: 'root'
})
export class AnswerService {
  private baseUrl = 'api/answer';

  constructor(private http: HttpClient, private messageService: MessageService) {
  }

  setAnswer(questionIndex: number, answerId: number): Observable<string> {
    const params = new HttpParams()
      .set('questionIndex', questionIndex.toString())
      .set('answerId', answerId.toString());

    return this.http.post<string>(`${this.baseUrl}/answers`, {}, {params: params})
      .pipe(
        catchError((response: Response) => {
          this.messageService.reportMessage(response);
          throw response;
        })
      );
  }

  getResult(): Observable<Result> {
    return this.http.get<Result>(`${this.baseUrl}/result`)
      .pipe(
        catchError((response: Response) => {
          this.messageService.reportMessage(response);
          throw response;
        })
      );
  }
}
